silkworm/rpc/core/estimate_gas_oracle.cpp

```diff
@@ -106,7 +105,7 @@ Task<intx::uint256> EstimateGasOracle::estimate_gas(const Call& call, const silk
             transaction.gas_limit = hi;
             result = try_execution(executor, block, transaction);
             SILK_DEBUG << "HI == cap tested again with " << (result.error_code == evmc_status_code::EVMC_SUCCESS ? "succeed" : "failed");
-        } else if (result.error_code == std::nullopt) {
+        } else if (result.pre_check_error_code == std::nullopt || *(result.pre_check_error_code) == PreCheckErrorCode::EC_INTRINSIC_GAS_TOO_LOW) {
```

Comment:
Probably this could be simplified to: `!result.pre_check_error_code || result.pre_check_error_code == ...`

Reply:
ok